### PR TITLE
define the interface of seed

### PR DESCRIPTION
**Pull request by wangforthinker**
Signed-off-by: allen.wq <allen.wq@alipay.com>

### Ⅰ. Describe what this PR did
define the seed interface. See https://github.com/dragonflyoss/Dragonfly/issues/1284.

### Ⅱ. Does this pull request fix one issue?

NONE.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
No need.

### CODE REVIEW

**Comment by hhhhsdxxxx** [comment]
lgtm!
Reviewed-by henry.hj@antfin.com

File: dfdaemon/seed/manager.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * limitations under the License.
+ */
+
+package seed
+
+import "time"
+
+const (
+	//  defaultDownloadConcurrency is the default concurrency limit of prefetching seed.
+	defaultDownloadConcurrency = 4
```
Comment (xujihui1985) [inline dfdaemon/seed/manager.go:23]: defaultDownloadConcurrency is unused at this moment, may be removed now?

Comment (wangforthinker) [reply dfdaemon/seed/manager.go]: fix it.

File: dfdaemon/seed/types.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	// the full length of seed file.
+	FullLength int64
+
+	// Seed will download data from rss which is divided by blocks.
+	// And block size is defined by BlockOrder. It should be limited [10, 31].
+	BlockOrder uint32
+
+	// expire time duration of seed file.
+	ExpireTimeDur time.Duration
```
[the remark this answers is not in the document]
Comment (wangforthinker) [reply dfdaemon/seed/types.go:44]: Good advice ! 
Add it now.

File: dfdaemon/seed/types.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	// filter these parameter via 'filter'. The usage of it is that different URL may generate the same
+	// download url.
+	Filters []string
+
+	// the full length of seed file.
+	FullLength int64
+
+	// Seed will download data from rss which is divided by blocks.
+	// And block size is defined by BlockOrder. It should be limited [10, 31].
+	BlockOrder uint32
```
Comment (lowzj) [inline dfdaemon/seed/types.go:41]: Why does it should be limited [10,31]?

Comment (wangforthinker) [reply dfdaemon/seed/types.go:41]: define the minimum and maximum of block size,  in the case, we may download from rss by block, so we consider the suitable block size to make the best  of bandwidth and reduce the cost of re-download.

File: dfdaemon/seed/seed.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ */
+
+package seed
+
+import "io"
+
+// Seed describes the seed file which represents the resource file defined by taskUrl.
+type Seed interface {
+	// Prefetch will start to download seed file to local cache.
+	Prefetch(perDownloadSize int64) (<-chan struct{}, error)
```
Comment (lowzj) [inline dfdaemon/seed/seed.go:24]: What's the meaning of `perDownloadSize` here?

Comment (wangforthinker) [reply dfdaemon/seed/seed.go:24]: when  a seed prefetch，`perDownloadSize ` is the maximum size of concurrent downloading for a seed.

**Comment by wangforthinker** [comment]
> The `taskID` is not in the `BaseInfo`, should it be contained?

Add it.

File: dfdaemon/seed/seed.go (hunk 4 above)
Comment (wangforthinker) [reply dfdaemon/seed/seed.go:24]: Add taskID in BaseInfo.

**Pull request merged**